misc/testerina/modules/testerina-core/src/main/java/org/ballerinalang/testerina/natives/io/BallerinaTypeCheck.java

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.ballerinalang.testerina.natives.io;
```

Comment:
Does BallerinaTypeCheck fit into natives.io? Cant it be considered part of core?

Reply:
+1 It would be better to move this to core.

Reply:
+1 It would be better to move this to core.

> Can we add tests and check the behavior for all types. specially anonymous types and other built in types.